applications/GeoMechanicsApplication/tests/cpp_tests/test_time_loop_executor.cpp

```diff
@@ -0,0 +1,356 @@
+        // intentionally empty
+    }
+    bool SolveSolutionStep()      override { return false; }
+    void FinalizeSolutionStep()   override {mCountFinalizeSolutionStepCalled += 1;}
+
+private:
+    TimeStepEndState::ConvergenceState mConvergenceState{TimeStepEndState::ConvergenceState::converged};
+    Model mModel;
+    ModelPart * mpModelPart = nullptr;
+    bool mIsCloned = false;
+    bool mIsRestoreCalled = false;
+    bool mIsSaveTotalDisplacementFieldCalled = false;
+    std::size_t mCountAccumulateTotalDisplacementFieldCalled = 0;
+    std::size_t mCountOutputProcessCalled = 0;
+    std::size_t mCountFinalizeSolutionStepCalled = 0;
+};
+
+class FixedCyclesTimeIncrementor : public TimeIncrementor
+{
+public:
+    explicit FixedCyclesTimeIncrementor(std::size_t DesiredNumOfCyclesPerStep)
+    : mNumCyclesPerStep{DesiredNumOfCyclesPerStep}
+    {}
+
+    [[nodiscard]] bool WantNextStep(const TimeStepEndState& rPreviousState) const override
+    {
+        return rPreviousState.time < mEndTime;
+    }
+    [[nodiscard]] bool WantRetryStep(std::size_t             CycleNumber,
+                                     const TimeStepEndState& rPreviousState) const override
+    {
+        return CycleNumber < mNumCyclesPerStep;
+    }
+    [[nodiscard]] double GetIncrement() const override
+    {
+        return 0.5;
+    }
+    void PostTimeStepExecution(const TimeStepEndState& rResultantState) override
+    {
+        // intentionally empty
+    }
+
+private:
+    std::size_t mNumCyclesPerStep;
+    double      mEndTime{1.0};
+};
+
+TimeStepEndState MakeConvergedStepState()
+{
+    auto result = TimeStepEndState{};
+    result.convergence_state = TimeStepEndState::ConvergenceState::converged;
+    return result;
+}
+
+std::shared_ptr<DummySolverStrategy> RunFixedCycleTimeLoop(std::size_t WantedNumOfCyclesPerStep)
+{
+    TimeLoopExecutor executor;
+    executor.SetTimeIncrementor(std::make_unique<FixedCyclesTimeIncrementor>(WantedNumOfCyclesPerStep));
+    auto solver_strategy = std::make_shared<DummySolverStrategy>(TimeStepEndState::ConvergenceState::converged);
+    executor.SetSolverStrategyTimeStepExecutor(solver_strategy);
+    const auto step_states = executor.Run(TimeStepEndState{});
+
+    return solver_strategy;
+}
+
+}
+
+namespace Kratos::Testing
+{
+
+KRATOS_TEST_CASE_IN_SUITE(TimeLoopReturnsPerformedStatesAfterRunningAnAlwaysConvergingSolverStrategy, KratosGeoMechanicsFastSuite)
+{
+    TimeLoopExecutor executor;
+    const auto increments = std::vector<double>{1.1, 1.2, 1.3, 1.4};
+    executor.SetTimeIncrementor(std::make_unique<PrescribedTimeIncrementor>(increments));
+    executor.SetSolverStrategyTimeStepExecutor(
+            std::make_unique<DummySolverStrategy>(TimeStepEndState::ConvergenceState::converged));
+    TimeStepEndState start_state;
+    start_state.convergence_state = TimeStepEndState::ConvergenceState::converged;
+    const auto step_states = executor.Run(start_state);
```

Comment:
We can simplify this to:
```suggestion
    const auto step_states = executor.Run(MakeConvergedStepState());
```

Reply:
Done